.cypress/integration/8_metrics_analytics.spec.js

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/// <reference types="cypress" />
+
+import {
+  delay,
+  PPL_METRICS,
+  PPL_METRICS_NAMES,
+  VIS_TYPE_LINE,
+  TESTING_PANEL,
+} from '../utils/metrics_constants';
+import { suppressResizeObserverIssue, COMMAND_TIMEOUT_LONG } from '../utils/constants';
+import { clearQuerySearchBoxText } from '../utils/event_analytics/helpers';
+import { landOnPanels } from '../utils/event_analytics/constants';
+
+const moveToMetricsHome = () => {
+  cy.visit(`${Cypress.env('opensearchDashboards')}/app/observability-metrics#/`);
+  cy.wait(delay * 3);
+};
+
+const moveToEventsExplorer = () => {
+  cy.visit(`${Cypress.env('opensearchDashboards')}/app/observability-logs#/explorer`);
+  cy.wait(delay * 3);
+};
+
+const moveToEventsHome = () => {
+  cy.visit(`${Cypress.env('opensearchDashboards')}/app/observability-logs#/`);
+  cy.wait(delay * 3);
+};
+
+describe('Creating custom metrics', () => {
+  beforeEach(() => {
+    moveToEventsExplorer();
+    clearQuerySearchBoxText('searchAutocompleteTextArea');
+    suppressResizeObserverIssue();
+  });
+
+  it('Create custom metric in event analytics and check it in events home', () => {
+    cy.get('[id^=autocomplete-textarea]').focus().type(PPL_METRICS[0], {
+      delay: 50,
+    });
+    cy.get('.euiButton__text').contains('Refresh').trigger('mouseover').click();
+    cy.wait(delay);
+    suppressResizeObserverIssue();
+    cy.get('button[id="main-content-vis"]').contains('Visualizations').trigger('mouseover').click();
+    cy.wait(delay * 2);
```

Comment:
Can we try to avoid explicit wait here and for other places as well?

Reply:
yeah will try to

Reply:
Will post a subsequent PR to eliminate wait time